test/cppsuite/test_harness/workload/thread_context.cxx

```diff
@@ -219,21 +219,29 @@ thread_context::update(scoped_cursor &cursor, uint64_t collection_id, const std:
 }
 
 bool
-thread_context::insert(scoped_cursor &cursor, uint64_t collection_id, uint64_t key_id)
+thread_context::insert(
+  scoped_cursor &cursor, uint64_t collection_id, uint64_t key_id, wt_timestamp_t ts)
+{
+    return insert(cursor, collection_id, key_to_string(key_id), ts);
+}
+
+bool
+thread_context::insert(
+  scoped_cursor &cursor, uint64_t collection_id, const std::string &key, wt_timestamp_t ts)
```

Comment:
Was wondering if we should also set ts = 0 here? just for styling purposes.

Reply:
The [Google style guide](https://google.github.io/styleguide/cppguide.html#Default_Arguments) does not say much about it. But it says the following that goes against what I have done though:
```
Default arguments are allowed on non-virtual functions when the default is guaranteed to always have the same value.
```
What do you think?

Reply:
If we follow this guideline, since we won't have always the same value, it means I would need to change the different calls to that function in the existing tests by passing a valid timestamp to `insert`. But then, if I add a timestamp to each `insert` call, we no longer need the default value...

Reply:
How I understand that sentence is that defaults are allowed on non-virtual function `insert()`. If the default is guaranteed to always be the same value. I think this is fine too because the default is always set to 0, unless you specifically supply a timestamp.

I was thinking if the definition has ts set to 0, we should also reflect that in the declaration as well.

Reply:
Hi @etienneptl and @jiechenbo , the code is correct in defining the default parameter ('= 0') in the declaration (ie in the `.h` file in this case). The `insert` methods are non-virtual, so default parameters are ok (default parameters are a very bad idea for virtual functions), and overloading them based on the presence or not of `ts` would be overkill, I think, in this case. I believe that if you try to add `= 0` to the `.cxx` file as well as the `.h` file you'll get a compiler error.

Reply:
Thanks @jeremythorp My mistake! Sorry for the confusion.